api/src/api/healthCheckRouter.ts

```diff
@@ -0,0 +1,61 @@
+import { Router } from "express";
+
+/* TODO: Building out this feature:
+ *
+ * - Ensure continued access to the /health endpoint for the entire app.
+ * - Create functions for checking Boomi and Dynamics health.
+ * - Use dependency inversion to make the health checks dynamic.
+ * - Wire up these functions to the appropriate endpoints.
+ */
+
+interface SuccessfulHealthCheckResponse {
+  success: true;
+  data: {
+    message: string;
+  };
+}
+
+interface UnsuccessfulHealthCheckResponse {
+  success: false;
+  error: {
+    timeout: boolean;
+    serverResponseCode?: number;
+    serverResponseBody?: string;
```

Comment:
[pebble] I like having `message` as a field in the Successful response because it lets individual handlers add context as needed- can we add that here as well?

Reply:
Yep! I'll add it. I'd love a way to indicate to others that it should be a safe return message that contains no PII, but would love some input on an appropriate name here. Don't want to accidentally leak info from our status checks somehow.

Reply:
Yeah that's a great point. I think `data` is the field name that would most likely tempt me to put userData in it. Maybe `responseContext`? 

Or if we want to be even stricter, we could have a type for possible responses (maybe `HealthCheckResponseMessage`). It would be slightly more annoying to add possible responses to the type, but you'd be looking at every previous example to see that its not user data. Plus, we'd likely have one response for 99% of healthy messages ("Alive"), so it would only be a few special cases where additional messages would be needed.

Reply:
Error -37. Haha. Yeah we can definitely do that. I'll tack onto any of the ones you think of here; and I'm definitely open to changing "Alive", as that was the pre-existing message but I have no loyalty to it.